### PR TITLE
Support of path completion in fish #327

### PR DESCRIPTION
**Pull request by volkov**
First version of absolute path completion for fish. #327 fix

### CODE REVIEW

**Comment by volkov** [comment]
@kislyuk @evanunderscore take a look please, I think it could be merged.

**Comment by evanunderscore** [comment]
Looks good to me. Are you able to add a test case to the test suite that verifies this? It could probably go in the base class since I'd expect this to work for Bash too.

File: README.rst [hunk 1]
```diff
@@ -290,7 +290,21 @@ By default help string is added as completion description.
...
+
+    register-python-argcomplete --shell fish /home/awesome-user/my-awesome-script | source
+
+then you can complete it by using ``/home/awesome-user/my-awesome-script`` or ``./my-awesome-script``.
+
+Unfortunately ``~/my-awesome-script`` would not work, to fix it you can run::
+
+    register-python-argcomplete --shell fish my-awesome-script -e /home/awesome-user/my-awesome-script | source
+
+This would enable completion for any ``my-awsome-script`` in any location.
```
Comment (kislyuk) [inline README.rst:307]:
```suggestion
This would enable completion for any ``my-awesome-script`` in any location.
```

**Review by kislyuk — COMMENTED** [review]
Thank you @volkov. As @evanunderscore said, this looks good but needs a regression test to make sure the new behavior is covered by the test suite.

**Comment by kislyuk** [comment]
@volkov let us know if you need help adding a test.

**Comment by volkov** [comment]
> Thank you @volkov. As @evanunderscore said, this looks good but needs a regression test to make sure the new behavior is covered by the test suite.

Ok, I will add first version of test to discuss.

**Comment by volkov** [comment]
I have added first version of test https://github.com/kislyuk/argcomplete/pull/359/files#diff-ae070fbd66c55f7bf108994c0a42ffe69eee7319b5bf0f47bef0fcdbd8b496b3R1385

> It could probably go in the base class since I'd expect this to work for Bash too.
I think its better to write separate base test for path completion because `setUp` is different:
* test directory should not be added to `PATH`
* completion should be registered only for path

**Comment by volkov** [comment]
Looks like bash don't resolve relative paths, same test for bash fails on `/prog basic f\t`

**Comment by evanunderscore** [comment]
Is it actually necessary to have a separate class for these tests? Having the test directory on the path shouldn't prevent you from trying to complete an absolute or relative path. If for some reason it is actually necessary to do that, I'd prefer to have the common logic in a base class rather than duplicating it.

I think it makes sense to split the absolute and relative tests into two test functions.

**Comment by volkov** [comment]
> Is it actually necessary to have a separate class for these tests?

Directory on the path is fine, but I can't test path completion when command completion registered.
When completion registered for command `prog` fish would trigger it also for `./prog` and for `test/prog`.
So I can't create test that would pass with both registered completions, but fail with only command completion.

> I'd prefer to have the common logic in a base class rather than duplicating it.
> I think it makes sense to split the absolute and relative tests into two test functions.

I'll try.

**Comment by volkov** [comment]
> So I can't create test that would pass with both registered completions, but fail with only command completion.

Actually it's possible if I have two scripts with different names, e.g `prog` and `frog` - first with command completion, second with path, but I don't want to duplicate test script.

**Comment by kislyuk** [comment]
LGTM, thanks for your patience @volkov.

**Pull request merged**